src/Tasks/CreateManifestResourceName.cs

```diff
@@ -296,86 +299,77 @@ private static bool IsValidEverettIdChar(char c)
-
-            // the first character has stronger restrictions than the rest
-            if (!IsValidEverettIdFirstChar(subName[0]))
-            {
-                // if the first character is not even a valid subsequent character, replace it with an underscore
-                if (!IsValidEverettIdChar(subName[0]))
+                // the first character has stronger restrictions than the rest
+                if (!IsValidEverettIdFirstChar(subName[0]))
                 {
-                    everettId.Append('_');
+                    // if the first character is not even a valid subsequent character, replace it with an underscore
+                    if (!IsValidEverettIdChar(subName[0]))
+                    {
+                        builder.Append('_');
+                    }
+                    // if it is a valid subsequent character, prepend an underscore to it
+                    else
+                    {
+                        builder.Append('_').Append(subName[0]);
+                    }
                 }
-                // if it is a valid subsequent character, prepend an underscore to it
                 else
                 {
-                    everettId.Append('_');
-                    everettId.Append(subName[0]);
+                    builder.Append(subName[0]);
                 }
-            }
-            else
-            {
-                everettId.Append(subName[0]);
-            }
 
-            // process the rest of the subname
-            for (int i = 1; i < subName.Length; i++)
-            {
-                if (!IsValidEverettIdChar(subName[i]))
+                // process the rest of the subname
+                for (int i = 1; i < subName.Length; i++)
                 {
-                    everettId.Append('_');
-                }
-                else
-                {
-                    everettId.Append(subName[i]);
+                    if (!IsValidEverettIdChar(subName[i]))
+                    {
+                        builder.Append('_');
+                    }
+                    else
+                    {
+                        builder.Append(subName[i]);
+                    }
                 }
             }
-
-            return everettId.ToString();
         }
 
         /// <summary>
         /// Make a folder name into an Everett-compatible identifier
         /// </summary>
-        internal static string MakeValidEverettFolderIdentifier(string name)
+        internal static void MakeValidEverettFolderIdentifier(StringBuilder builder, string name)
         {
             ErrorUtilities.VerifyThrowArgumentNull(name, nameof(name));
 
-            // give string length to avoid reallocations; +1 since the resulting string may be one char longer than the
-            // original - if the name is a single underscore we add another underscore to it
-            var everettId = new StringBuilder(name.Length + 1);
+            if (!string.IsNullOrEmpty(name))
```

Comment:
Code styling nit: I couldn't find anything specific over at the [coding style document](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md) we refer to in our contributing guide, but I suggest turning this into something like `if(string.IsNullOrEmpty(name)) { return; }` to avoid nesting.